### PR TITLE
Be authorization with gmail api 12

### PR DESCRIPTION
**Pull request by Patton-L**
**What it does**
Complete backend google auth logic.
Integrated backend and frontend google auth routine.

**What to look out for**
A example test page to show google auth succeed.

**What to avoid**
The gmail_auth_handler test will only work without JWT and needs manual input; This can be done later by mock Google oauth server.
There is no logic at front-end page to avoid invalid google oauth return from server, this will be fix later.

### CODE REVIEW

File: client/src/components/OauthCallback.jsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
+// this component will eventually be rendered in the Dashboard page
+
+import React from "react";
+import GoogleButton from "react-google-button";
```
Comment (mcgraths7) [inline client/src/components/OauthCallback.jsx]: Remove unused files

Are you using eslint?

**Review by mcgraths7 — CHANGES_REQUESTED** [review]
Looks good, I would just clean up some of the unused files before merging

File: client/src/components/OauthCallback.jsx (hunk 1 above)
Comment (Patton-L) [reply client/src/components/OauthCallback.jsx]: Done

**Review by mcgraths7 — APPROVED** [review]
👍

File: server/credentials.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{"web":{"client_id":"542989482020-ed1pfj7ahps0qgn3fmbu8eoprvfjiifi.apps.googleusercontent.com","project_id":"quickstart-1603250317186","auth_uri":"https://accounts.google.com/o/oauth2/auth","token_uri":"https://oauth2.googleapis.com/token","auth_provider_x509_cert_url":"https://www.googleapis.com/oauth2/v1/certs","client_secret":"X_a9SleZL_tjXsPaX9GC2X7D","redirect_uris":["https://localhost:3000/t"]}}
```
Comment (skassam21) [inline server/credentials.json:1]: We should gitignore this!

File: server/api/gmail_auth_handler.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    'https://www.googleapis.com/auth/userinfo.profile',
+    'https://www.googleapis.com/auth/gmail.compose',
+    'https://www.googleapis.com/auth/userinfo.email',
+    # Add other requested scopes.
+]
+
+CLIENTSECRETS_LOCATION = './credentials.json' # TODO: goes to config file
+# url_for('gmail_auth_handler.gmail_oauth_callback', _external=True)
+REDIRECT_URI = 'http://localhost:5000/gmail_oauth_callback'
+REDIRECT_URI_FRONT = 'http://localhost:3000/gmail_oauth_callback'
```
Comment (skassam21) [inline server/api/gmail_auth_handler.py:25]: can we make the beginning of these URIs (the domains) environment variables so we can change this later?

**Pull request merged**